src/pages/RFOX/hooks/useRfoxClaimActionSubscriber.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { useGetUnstakingRequestsQuery } from './useGetUnstakingRequestsQuery'
+
+import { bn } from '@/lib/bignumber/bignumber'
+import { fromBaseUnit } from '@/lib/math'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import { selectPendingRfoxClaimActions } from '@/state/slices/actionSlice/selectors'
+import { ActionStatus, ActionType } from '@/state/slices/actionSlice/types'
+import { selectAssets } from '@/state/slices/selectors'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+// Note: the is a slight misnomer here compared to what we conventionally call a subscriber in action center (a poller)
+// However, this *is* reactive on useGetUnstakingRequestsQuery(), with inner queries which do get invalidated, so it is a subscriber in a way, just different
+
+export const useRfoxClaimActionSubscriber = () => {
+  const translate = useTranslate()
+  const dispatch = useAppDispatch()
+  const assets = useAppSelector(selectAssets)
+
+  const allUnstakingRequests = useGetUnstakingRequestsQuery()
+
+  const pendingRfoxClaimActions = useAppSelector(selectPendingRfoxClaimActions)
+  const actionIds = useAppSelector(actionSlice.selectors.selectActionIds)
+
+  useEffect(() => {
+    if (!pendingRfoxClaimActions.length) return
+    const now = Date.now()
+
+    pendingRfoxClaimActions.forEach(action => {
+      if (!action.rfoxClaimActionMetadata.txHash) return
+      const asset = assets[action.rfoxClaimActionMetadata.request.stakingAssetId]
+      if (!asset) return
```

Comment:
Maybe we should early return if the action is already status  `Claimed`? Just in case as a safety

Reply:
I am confusion @NeOMakinG - `selectPendingRfoxClaimActions` selects `ActionStatus.Pending` actions, so this can't be `ActionStatus.Claimed`